Solution1/Solution1.Tests/BL/Services/WeatherServiceTests.cs

```diff
@@ -165,40 +171,62 @@ public async Task GetWeatherByArrayCityNameAsync_ReturnedDictionaryForWeatherRes
                  .ReturnsAsync(validResult);
 
             // Act
-            var result = await _weatherService.GetWeatherByArrayCityNameAsync(listCityName);
-
+            var result = await _weatherService.GetWeatherByArrayCityNameAsync(listCityName, CancellationToken.None);
+            
             // Assert
-            Assert.Contains(result.Keys, k => k == true);
-            Assert.Contains(result.Keys, k => k == false);
+            Assert.Contains(result.Keys, k => k == ResponseStatus.Successful);
+            Assert.Contains(result.Keys, k => k == ResponseStatus.Fail);
 
-            Assert.Equal(2, result[true].Count());
-            Assert.Equal(3, result[false].Count());
+            Assert.Equal(2, result[ResponseStatus.Successful].Count());
+            Assert.Equal(3, result[ResponseStatus.Fail].Count());
 
-            Assert.Contains(result[true], weatherResponse => weatherResponse.CityName == cityName);
-            Assert.Contains(result[true], weatherResponse => weatherResponse.CityName == cityName2);
-            Assert.Contains(result[false], weatherResponse => weatherResponse.CityName == cityName3 
-                                                              && weatherResponse.ErrorMessage == errorMessage3); 
-            Assert.Contains(result[false], weatherResponse => weatherResponse.CityName == cityName4 
+            Assert.Contains(result[ResponseStatus.Successful], weatherResponse => weatherResponse.CityName == cityName);
+            Assert.Contains(result[ResponseStatus.Successful], weatherResponse => weatherResponse.CityName == cityName2);
+            Assert.Contains(result[ResponseStatus.Fail], weatherResponse => weatherResponse.CityName == cityName3
+                                                              && weatherResponse.ErrorMessage == errorMessage3);
+            Assert.Contains(result[ResponseStatus.Fail], weatherResponse => weatherResponse.CityName == cityName4
                                                               && weatherResponse.ErrorMessage == errorMessage4);
-            Assert.Contains(result[false], weatherResponse => weatherResponse.CityName == cityName5 
+            Assert.Contains(result[ResponseStatus.Fail], weatherResponse => weatherResponse.CityName == cityName5
                                                               && weatherResponse.ErrorMessage == errorMessage5);
         }
 
+        [Fact]
+        public async Task GetForecastByCityNameAsync_GenerateOperationCanceledException_Success()
+        {
+            // Arrange
+            var cityName2 = "Paris";
+            var errorMessage = "Timeout exceeded";
+            var listCityName = new List<string>() { cityName, cityName2 };
+            
+            //Act
+            var result = await _weatherService.GetWeatherByArrayCityNameAsync(listCityName, new CancellationToken(true));
+
+            // Assert
+            Assert.Contains(result.Keys, k => k == ResponseStatus.Canceled);
```

Comment:
what if result is null? or Keys is null?

Reply:
+